src/gap_to_julia.jl

```diff
@@ -133,15 +133,15 @@ gap_to_julia(::Type{T}, x::GapInt) where {T<:Integer} = T(x)
 gap_to_julia(::Type{Rational{T}}, x::GapInt) where {T<:Integer} = Rational{T}(x)
 
 ## Floats
-gap_to_julia(::Type{T}, obj::GapObj) where {T<:AbstractFloat} = T(obj)
+gap_to_julia(::Type{T}, obj::GapObj) where {T<:AbstractFloat} = applicable(T, obj) ? T(obj) : T(Float64(obj))
 
 ## Chars
 gap_to_julia(::Type{Char}, obj::GapObj) = Char(obj)
 gap_to_julia(::Type{Cuchar}, obj::GapObj) = Cuchar(obj)
 
 ## Strings
 gap_to_julia(::Type{String}, obj::GapObj) = String(obj)
-gap_to_julia(::Type{T}, obj::GapObj) where {T<:AbstractString} = T(obj)
+gap_to_julia(::Type{T}, obj::GapObj) where {T<:AbstractString} = applicable(T, obj) ? T(obj) : T(String(obj))
```

Comment:
```suggestion
gap_to_julia(::Type{String}, obj::GapObj) where {T<:AbstractString} = String(obj)
```

Reply:
I reworked most of this PR and included this change.